src/app/WriteClient.h

```diff
@@ -118,25 +123,81 @@ class WriteClient : public Messaging::ExchangeDelegate
                 const Optional<uint16_t> & aTimedWriteTimeoutMs) :
         mpExchangeMgr(apExchangeMgr),
         mpCallback(apCallback), mTimedWriteTimeoutMs(aTimedWriteTimeoutMs)
-    {}
+    {
+        CreateNewMessage();
```

Comment:
This shouldn't be happening in the constructor - it should be happening in `EncodeAttributeWritePayload` (and other similar methods) if it detects we haven't started a message yet.